src/transformStateWithClones.js

```diff
@@ -8,6 +8,23 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+    for (const key in actions) {
+    switch (actions[key].type) {
+      case 'addProperties':
+        for (const extraKey in actions[key].extraData) {
+          state[extraKey] = actions[key].extraData[extraKey];
+        }
+        break;
+      case 'removeProperties':
+        for (const extraKey in actions[key].keysToRemove) {
+          delete state[actions[key].keysToRemove[extraKey]];
+        }
+        break;
+      case 'clear':
+        Object.keys(state).forEach(currentKey => delete state[currentKey]);
```

Comment:
When clearing the state, make sure to create a new object instead of modifying the existing one directly.